File: sportsbook-app/payment-contracts/src/main/java/com/sportsbook/payment/contracts/DepositEnrichedEvent.java
package com.sportsbook.payment.contracts;

public record DepositEnrichedEvent(
        DepositInitiatedEvent original,
        DepositRiskFlag riskFlag,
        int recentCreditBlocks,
        int velocityCountLastHour
) {
    public static DepositEnrichedEvent noRisk(DepositInitiatedEvent original) {
        return new DepositEnrichedEvent(original, DepositRiskFlag.NONE, 0, 0);
    }
}

File: sportsbook-app/payment-contracts/src/main/kotlin/com/sportsbook/payment/contracts/DepositEnrichedEvent.kt
package com.sportsbook.payment.contracts

data class DepositEnrichedEvent(
    val original: DepositInitiatedEvent,
    val riskFlag: DepositRiskFlag,
    val recentCreditBlocks: Int,
    val velocityCountLastHour: Int,
) {
    companion object {
        @JvmStatic
        fun noRisk(original: DepositInitiatedEvent): DepositEnrichedEvent =
            DepositEnrichedEvent(original, DepositRiskFlag.NONE, 0, 0)
    }
}
